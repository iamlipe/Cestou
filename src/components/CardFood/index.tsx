import React, {memo} from 'react';
import styled from 'styled-components/native';
import {useTranslation} from 'react-i18next';

import {translateFoodToPortuguese} from '@/helpers/translate';

import Counter from '../Counter';

interface Props {
  name: string;
  control: any;
  maxQuantity: number;
  image: string;
}
const CardFood = ({name, image, control, maxQuantity}: Props) => {
  const {t, i18n} = useTranslation();

  return (
    <StyledContainer>
      <StyledImage source={{uri: image}} />

      <StyledContainerInfo>
        <StyledTitleCard>
          {i18n.language === 'pt' ? translateFoodToPortuguese(name) : name}
        </StyledTitleCard>
        <StyledButton>
          <StyledTextLink>{t('Text.ComponentCardFood.SeeMore')}</StyledTextLink>
        </StyledButton>
      </StyledContainerInfo>
      <Counter name={name} control={control} maxQuantity={maxQuantity} />
    </StyledContainer>
  );
};

const StyledContainer = styled.View`
  height: 72px;
  flex-direction: row;
  align-items: center;
  border: 1px solid ${({theme}) => theme.colors.GRAY_100};
  border-radius: 5px;
  margin-bottom: 16px;
`;

const StyledImage = styled.Image`
  height: 100%;
  width: 30%;
  border-bottom-left-radius: 4px;
  border-top-left-radius: 4px;
`;

const StyledContainerInfo = styled.View`
  width: 40%;
  height: 80%;
  flex-direction: column;
  justify-content: space-evenly;

  padding-left: 8px;
`;

const StyledTitleCard = styled.Text`
  font-family: ${({theme}) => theme.fonts.SEMIBOLD_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.SMALLER};
  color: ${({theme}) => theme.colors.GRAY_900};
`;

export const StyledButton = styled.TouchableOpacity``;

export const StyledTextLink = styled.Text`
  font-family: ${({theme}) => theme.fonts.REGULAR_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.SMALLEST};
  color: ${({theme}) => theme.colors.PRIMARY_800};
  text-decoration: underline;
`;

export default memo(CardFood);
