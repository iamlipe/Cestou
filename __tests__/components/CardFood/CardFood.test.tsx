import React from 'react';
import {
  renderWithThemeProvider,
  matchSnapshotWithProvider,
} from '@/config/functions/Testing';

import CardFood from '@/components/CardFood';

jest.mock('react-hook-form', () => ({
  useController: () => ({
    field: {
      onChange: jest.fn(),
    },
  }),
}));

describe('CardFood', () => {
  const name = 'test';
  const control = jest.fn();
  const maxQuantity = 3;
  const image =
    'https://s3.filebase.com/ioasys-g5-fome-zero/foods/fruits-food.png';

  test('should render correctly', () => {
    renderWithThemeProvider(
      <CardFood
        name={name}
        control={control}
        maxQuantity={maxQuantity}
        image={image}
      />,
    );
  });

  test('should matches snapshot', () => {
    matchSnapshotWithProvider(
      <CardFood
        name={name}
        control={control}
        maxQuantity={maxQuantity}
        image={image}
      />,
    );
  });
});
