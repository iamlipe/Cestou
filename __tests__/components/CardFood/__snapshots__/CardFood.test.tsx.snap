// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardFood should matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#F2F2F8",
        "borderRadius": 5,
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 72,
        "marginBottom": 16,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://s3.filebase.com/ioasys-g5-fome-zero/foods/fruits-food.png",
      }
    }
    style={
      Array [
        Object {
          "borderBottomLeftRadius": 4,
          "borderTopLeftRadius": 4,
          "height": "100%",
          "width": "30%",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
          "height": "80%",
          "justifyContent": "space-evenly",
          "paddingLeft": 8,
          "width": "40%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1A1A1A",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 27,
          },
        ]
      }
    >
      test
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#00843F",
              "fontFamily": "SourceSansPro-Regular",
              "fontSize": 24,
              "textDecorationColor": "black",
              "textDecorationLine": "underline",
              "textDecorationStyle": "solid",
            },
          ]
        }
      >
        Ver exemplos
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#C1E9CF",
          "borderColor": "#00B761",
          "borderRadius": 16,
          "borderStyle": "solid",
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 32,
          "justifyContent": "space-between",
          "paddingBottom": 0,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 0,
          "width": 86,
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#1A1A1A",
              "fontSize": 14,
            },
            undefined,
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="decrement-button"
      >
        
      </Text>
    </View>
    <TextInput
      editable={false}
      style={
        Array [
          Object {
            "color": "#1A1A1A",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 27,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
            "width": 10,
          },
        ]
      }
      testID="counter-test"
      value="3"
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#1A1A1A",
              "fontSize": 14,
            },
            undefined,
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="increment-button"
      >
        
      </Text>
    </View>
  </View>
</View>
`;
